src/simplify_algebra.cpp

```diff
@@ -421,15 +436,22 @@ struct find_dot_mul
 struct find_mul_add
 {
     auto matcher() const
-    {
-        return match::name("mul")(match::either_arg(0, 1)(
-            match::name("add")(
-                match::either_arg(0, 1)(
-                    match::any().bind("x"),
-                    match::any_of(conv_const_weights(), match::is_constant()).bind("b")),
-                match::none_of(match::args(match::is_constant(), match::is_constant())),
-                match::used_once()),
-            match::is_constant().bind("a")));
+    {   
```

Comment:
Please add a comment(s) here in this matcher, to indicate what is exactly being done. Are you bypassing `slice` followed by another `slice`, it isn't clear to me.